components/org.wso2.carbon.identity.api.user.authorized.apps/org.wso2.carbon.identity.rest.api.user.authorized.apps.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/authorized/apps/v2/utils/AuthorizedAppsServicesHolder.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.rest.api.user.authorized.apps.v2.utils;
+
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.oauth.OAuthAdminServiceImpl;
+import org.wso2.carbon.identity.oauth2.OAuth2ScopeService;
+import org.wso2.carbon.user.core.service.RealmService;
+
+/**
+ * This class holds the service instances required for authorized apps management.
+ */
+public class AuthorizedAppsServicesHolder {
+
+    private AuthorizedAppsServicesHolder() {
+
+    }
+
+    private static class ApplicationManagementServiceHolder {
+
+        static final ApplicationManagementService SERVICE = (ApplicationManagementService) PrivilegedCarbonContext
+                .getThreadLocalCarbonContext().getOSGiService(ApplicationManagementService.class, null);
```

Comment:
shouldn't these have the same null check as the realm service

Reply:
https://github.com/wso2/identity-api-user/pull/246/files#diff-8b236739fd2feafc6bdd0f8ab0a14a02f11ab2d1b4e26306677445874a62e60aR75

null check is being handled from here.